src/Symfony/Component/Validator/Mapping/Loader/AutoMappingTrait.php

```diff
@@ -36,6 +36,6 @@ private function isAutoMappingEnabledForClass(ClassMetadata $metadata, string $c
         }
 
         // Fallback on the config
-        return null === $classValidatorRegexp || preg_match($classValidatorRegexp, $metadata->getClassName());
+        return null !== $classValidatorRegexp && preg_match($classValidatorRegexp, $metadata->getClassName());
```

Comment:
⚠️ This is a BC break for the ones using the component directly:
- before, setting `$classValidatorRegexp` to `null` would always consider the auto-mapping enabled for class by default
- now, `null` will consider auto-mapping disabled for class by default, but can still be enabled by using `EnableAutoMapping`.

This needs a proper BC layer in any case, but I think the current way it works is not the most relevant. If you want it to be enabled by default for all classes, just use `".*"`. `null` should consider the auto-mapping feature not enabled for the class.

Reply:
What about just doing the BC break? The trait is new to 4.4.0, let's handle this as a bug.

Reply:
That's not about the trait actually. That's about 4.3 behavior of Doctrine/PropertyInfo loaders with a `null` regex. On 4.3 it was expected to enable the automapping with a `null` regex.
Or do you mean handle this as a bug in 4.3?

Reply:
/cc @dunglas WDYT?

Reply:
see also #33828

Reply:
this behavior already changed several times so... I'm in favor of introducing this change now, even if it's a real BC break (we'll have to document this in UPGRADE.md).